src/network/dns/resolver.rs

```diff
@@ -11,8 +14,26 @@ pub trait Lookup {
 pub struct Resolver(TokioAsyncResolver);
 
 impl Resolver {
-    pub async fn new(runtime: Handle) -> Result<Self, failure::Error> {
-        let resolver = TokioAsyncResolver::from_system_conf(runtime).await?;
+    pub async fn new(
+        runtime: Handle,
+        dns_server: &Option<Ipv4Addr>,
+    ) -> Result<Self, failure::Error> {
+        let resolver = match dns_server {
+            Some(dns_server_address) => {
+                let mut config = ResolverConfig::new();
+                let options = ResolverOpts::default();
+                // let socket = SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(1, 1, 1, 1), 53));
```

Comment:
Should this comment be removed? Is it something that still needs addressing?

Reply:
Hot damn, thanks for catching this :)